leaders/src/main/java/edu/cornell/partitioner/ClassPartitioner.java

```
@@ -1,52 +1,102 @@
+            for (String test : testsList) {
+                JsonObject testClass = new JsonObject();
+                testClass.addProperty("name", test);
+                testClasses.add(testClass);
+            }
+            root.add("testclasses", testClasses);
+
+            // Convert JsonObject to String for the POST request body
+            String jsonBody = gson.toJson(root);
+
+            // Build HttpRequest with JSON body
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(url))
+                    .header("Content-Type", "application/json")
+                    .POST(HttpRequest.BodyPublishers.ofString(jsonBody))
+                    .build();
+
+            // Send HttpRequest and receive HttpResponse
+            HttpResponse<String> response = client.send(request,
+                    HttpResponse.BodyHandlers.ofString());
+
+            // Parse JSON response to extract partitions
+            JsonObject jsonResponse = gson.fromJson(response.body(), JsonObject.class);
+            JsonArray partitions = jsonResponse.getAsJsonArray("partitions");
+            List<Set<String>> partitionsList = new ArrayList<>();
+
+            for (int i = 0; i < partitions.size(); i++) {
+                JsonObject partition = partitions.get(i).getAsJsonObject();
+                JsonArray testClassesArray = partition.getAsJsonArray("testclasses");
+                Set<String> testSet = new HashSet<>();
+
+                for (int j = 0; j < testClassesArray.size(); j++) {
+                    JsonObject testClassObject = testClassesArray.get(j).getAsJsonObject();
+                    testSet.add(testClassObject.get("name").getAsString());
+                }
+
+                partitionsList.add(testSet);
+            }
+
+            return partitionsList;
+
+        } catch (IOException | InterruptedException e) {
+            LOGGER.error(e.toString());
+            return null; // or handle error appropriately
```

I'm thinking we might wait to propagate this exception instead of catching it, logging it, and returning null. We still want the client of this method to handle an exception, so it might just be easier to declare that we throw `IOException` and `InterruptedException`.